backend/pkg/httpserver/create_saved_search.go

```diff
@@ -61,28 +61,64 @@ func (f fieldValidationErrors) hasErrors() bool {
 	return len(f.fieldErrorMap) > 0
 }
 
-func validateSavedSearch(input *backend.SavedSearch) *fieldValidationErrors {
-	fieldErrors := &fieldValidationErrors{fieldErrorMap: nil}
+// validateSavedSearchName checks the validity of the saved search name.
+// It expects a pointer to handle potential nil values during updates.
+func validateSavedSearchName(name *string, fieldErrors *fieldValidationErrors) {
+	if name == nil {
+		// Treat nil as invalid if validation is requested
+		fieldErrors.addFieldError("name", errSavedSearchInvalidNameLength)
 
-	if len(input.Name) < savedSearchNameMinLength || len(input.Name) > savedSearchNameMaxLength {
+		return
+	}
+	if len(*name) < savedSearchNameMinLength || len(*name) > savedSearchNameMaxLength {
 		fieldErrors.addFieldError("name", errSavedSearchInvalidNameLength)
 	}
+}
+
+// validateSavedSearchQuery checks the validity of the saved search query.
+// It expects a pointer to handle potential nil values during updates.
+func validateSavedSearchQuery(query *string, fieldErrors *fieldValidationErrors) {
+	if query == nil {
+		fieldErrors.addFieldError("query", errSavedSearchInvalidQueryLength)
+
+		return
+	}
 
-	if len(input.Query) < savedSearchQueryMinLength || len(input.Query) > savedSearchQueryMaxLength {
+	if len(*query) < savedSearchQueryMinLength || len(*query) > savedSearchQueryMaxLength {
```

Comment:
```suggestion
	if len(*query) < savedSearchQueryMinLength || len(*query) > savedSearchQueryMaxLength {
		fieldErrors.addFieldError("query", errSavedSearchInvalidQueryLength)
		
		return
	} 
	
	// Only parse if length is okay
	parser := searchtypes.FeaturesSearchQueryParser{}
	_, err := parser.Parse(*query)
	if err != nil {
		fieldErrors.addFieldError("query", errQueryDoesNotMatchGrammar)
	}
```
Optional nit: return early and remove nested else if loop

Reply:
If we return early, it won't be able to also validate the description field below.

Reply:
Edit: my previous comment is wrong. I was thinking about the old structure. I will do this too

Reply:
why is that the case? it would return regardless outside of the if statement

Reply:
Just saw the comment. Thanks!

Reply:
No worries! Done. Thanks for the review.